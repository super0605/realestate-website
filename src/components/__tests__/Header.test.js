import * as React from "react";
import { mount, shallow } from "enzyme";
import HeaderComponent from "../Header";

describe("Header Component", () => {
  let props;
  let mountedHeaderComponent;
  let wrapper;

  const renderWrapper = (bShallow) => {
    if (!mountedHeaderComponent) {
      mountedHeaderComponent = bShallow
        ? shallow(<HeaderComponent {...props} />)
        : mount(<HeaderComponent {...props} />);
    }
    return mountedHeaderComponent;
  };

  beforeEach(() => {
    wrapper = renderWrapper(true);
  });

  it("should render without problems", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
