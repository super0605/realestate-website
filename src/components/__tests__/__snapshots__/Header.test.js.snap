// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render without problems 1`] = `
<Styled(Header)>
  <styled.div>
    <styled.div
      onClick={[Function]}
    >
      <ForwardRef(MenuOutlined) />
    </styled.div>
    <styled.div>
      KASAZ
    </styled.div>
  </styled.div>
  <styled.div>
    <SearchInput />
  </styled.div>
  <styled.div>
    <Menu
      defaultSelectedKeys={
        Array [
          "1",
        ]
      }
      mode="horizontal"
      theme="light"
    >
      <Styled(MenuItem)
        key="1"
      >
        SERVICES
      </Styled(MenuItem)>
      <Styled(MenuItem)
        key="2"
      >
        GUIDES
      </Styled(MenuItem)>
      <Styled(MenuItem)
        key="3"
      >
        BLOG
      </Styled(MenuItem)>
      <Styled(MenuItem)
        key="4"
      >
        LOG IN
      </Styled(MenuItem)>
    </Menu>
  </styled.div>
</Styled(Header)>
`;
